gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureJavaServiceDependencies.java

```diff
@@ -42,13 +44,18 @@ static void configureJavaServiceDependencies(
     }
 
     private static Set<ProductDependency> convertDependencies(Set<ServiceDependency> serviceDependencies) {
+        Preconditions.checkArgument(
+                serviceDependencies.size() != 1
+                        || !Iterables.getOnlyElement(serviceDependencies).getOptional(),
+                "a single optional service dependency is not supported");
```

Comment:
Can we add a comment here with a short summery of the reason you outlined in the PR comment? Otherwise in the future, it might be hard to rediscover why we don't allow a single optional service dep.

Reply:
Done, thanks!